app/models/manageiq/providers/ansible_tower/configuration_manager/refresh_parser.rb

```diff
@@ -68,9 +68,17 @@ def parse_host(host)
           :virtual_instance_ref => host.instance_id,
         }
 
+        cross_link_host(new_result)
+
         return uid, new_result
       end
 
+      def cross_link_host(new_result)
+        vm = Vm.where(:uid_ems => new_result[:virtual_instance_ref]).first
+        new_result[:counterpart_id]   = vm.try(:id)
+        new_result[:counterpart_type] = vm.class.try(:base_class).to_s.presence
```

Comment:
This feels like it should just be `new_result[:counterpart] = vm`...you shouldn't need to mess with the id/type fields directly.